### PR TITLE
[Asset Inventory][Azure] Add storage assets

### PR DESCRIPTION
**Pull request by kubasobon**
### Summary of your changes

Add 3 storage assets: Queues, Queue Services, and Blob Services.

### Related Issues

Closes https://github.com/elastic/security-team/issues/10162
Closes https://github.com/elastic/security-team/issues/10163

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: internal/inventory/asset.go [hunk 1]
```diff
@@ -99,6 +101,9 @@ const (
 	SubTypeAzureSQLServer           AssetSubType = "azure-sql-server"
 	SubTypeAzureSnapshot            AssetSubType = "azure-snapshot"
 	SubTypeAzureStorageAccount      AssetSubType = "azure-storage-account"
+	SubTypeAzureStorageBlobService  AssetSubType = "azure-storage-blob-service"
+	SubTypeAzureStorageQueue        AssetSubType = "azure-storage-queue"
+	SubTypeAzureStorageQueueService AssetSubType = "azure-storage-queue-service"
```
Comment (romulets) [inline internal/inventory/asset.go:106]: What is the difference between azure-storage-queue and azure-storage-queue-service?

File: internal/inventory/azurefetcher/fetcher_storage.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	storageAccounts, err := f.provider.ListStorageAccounts(ctx, subscriptionIds)
+	if err != nil {
+		return nil, fmt.Errorf("error listing storage accounts: %v", err)
+	}
+
+	return storageAccounts, nil
+}
+
+func (f *storageFetcher) fetch(ctx context.Context, storageAccounts []azurelib.AzureAsset, resourceName string, function storageProviderFunc, classification inventory.AssetClassification, assetChan chan<- inventory.AssetEvent) {
+	f.logger.Infof("Fetching %s", resourceName)
```
Comment (romulets) [inline internal/inventory/azurefetcher/fetcher_storage.go:95]: Do we have this for every fetcher? I wonder if we are not over-logging

File: internal/resources/providers/azurelib/inventory/storage_provider.go [hunk 3]
```diff
@@ -152,6 +180,62 @@ func (p *storageAccountProvider) ListStorageAccountBlobServices(ctx context.Cont
...
+				TenantId:       sa.TenantId,
+				Properties:     properties,
+				Extension: map[string]any{
+					ExtensionStorageAccountID:   sa.Id,
+					ExtensionStorageAccountName: sa.Name,
+				},
+			})
+		}
+		if errs != nil {
+			return nil, fmt.Errorf("error while transforming azure queue services for storage accounts %s: %w", sa.Id, errs)
```
Comment (romulets) [inline internal/resources/providers/azurelib/inventory/storage_provider.go:232]: I wonder if we should return nil here and don't store any StorageQueueService. 

The error is not a blocking one, right? We rather have the asset without properties anyway - and log the error.

Wdyt?

File: internal/resources/providers/azurelib/inventory/storage_provider.go [hunk 4]
```diff
@@ -152,6 +180,62 @@ func (p *storageAccountProvider) ListStorageAccountBlobServices(ctx context.Cont
...
+func (p *storageAccountProvider) ListStorageAccountQueues(ctx context.Context, storageAccounts []AzureAsset) ([]AzureAsset, error) {
+	var assets []AzureAsset
+	for _, sa := range storageAccounts {
+		responses, err := p.client.AssetQueues(ctx, sa.SubscriptionId, nil, sa.ResourceGroup, sa.Name, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while fetching azure queues for storage accounts %s: %w", sa.Id, err)
+		}
+
+		queues, err := transformQueues(responses, sa)
+		if err != nil {
```
Comment (romulets) [inline internal/resources/providers/azurelib/inventory/storage_provider.go:192]: Same comment as StorageQueueService, should we stop the execution if the transform queue can't extract properties?

**Review by romulets — CHANGES_REQUESTED** [review]

File: internal/inventory/azurefetcher/fetcher_storage.go (hunk 2 above)
Comment (kubasobon) [reply internal/inventory/azurefetcher/fetcher_storage.go:95]: Yes we do. Assuming we'll have ~200 assets, even if we're scanning every minute, that's 400 lines/minute. I'd say that is a very tame amount of logs. If we want to change the approach, then let's do it in a separate PR for all of the fetchers.

File: internal/resources/providers/azurelib/inventory/storage_provider.go (hunk 3 above)
Comment (kubasobon) [reply internal/resources/providers/azurelib/inventory/storage_provider.go:232]: Makes sense 👍

File: internal/resources/providers/azurelib/inventory/storage_provider.go (hunk 4 above)
Comment (kubasobon) [reply internal/resources/providers/azurelib/inventory/storage_provider.go:192]: Makes sense 👍

**Review by romulets — APPROVED** [review]

**Pull request merged**